src/storage/postgres/src/main/java/org/locationtech/geogig/storage/postgresql/PGCache.java

```diff
@@ -0,0 +1,214 @@
+/* Copyright (c) 2017 Boundless and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/org/documents/edl-v10.html
+ *
+ * Contributors:
+ * Gabriel Roldan (Boundless) - initial implementation
+ */
+package org.locationtech.geogig.storage.postgresql;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.locationtech.geogig.model.ObjectId;
+import org.locationtech.geogig.model.RevObject;
+import org.locationtech.geogig.storage.ConfigDatabase;
+import org.locationtech.geogig.storage.datastream.LZ4SerializationFactory;
+import org.locationtech.geogig.storage.datastream.v2_2.DataStreamSerializationFactoryV2_2;
+import org.locationtech.geogig.storage.impl.ObjectSerializingFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.RemovalListener;
+import com.google.common.cache.RemovalNotification;
+import com.google.common.cache.Weigher;
+
+public class PGCache {
+
+    private static final ObjectSerializingFactory ENCODER = new LZ4SerializationFactory(//
+            DataStreamSerializationFactoryV2_2.INSTANCE//
+    );
+
+    protected static final int ESTIMATED_OBJECTID_SIZE = 28;
+
+    private ObjectSerializingFactory encoder = ENCODER;
+
+    private static Weigher<ObjectId, byte[]> weigher = new Weigher<ObjectId, byte[]>() {
+
+        // private static Instrumentation instrumentation;
+
+        @Override
+        public int weigh(ObjectId key, byte[] value) {
+            return ESTIMATED_OBJECTID_SIZE + value.length;
+        }
+
+    };
+
+    private static class SizeTracker implements RemovalListener<ObjectId, byte[]> {
+
+        public final AtomicLong size = new AtomicLong();
+
+        @Override
+        public void onRemoval(RemovalNotification<ObjectId, byte[]> notification) {
+            ObjectId key = notification.getKey();
+            byte[] value = notification.getValue();
+            int weigh = weigher.weigh(key, value);
+            size.addAndGet(-weigh);
+        }
+
+        public void inserted(ObjectId id, byte[] value) {
+            int weigh = weigher.weigh(id, value);
+            size.addAndGet(weigh);
+        }
+    }
+
+    public static PGCache build(ConfigDatabase configdb) {
+        Optional<Long> maxSize = configdb.get(Environment.KEY_ODB_BYTE_CACHE_MAX_SIZE, Long.class);
```

Comment:
since the cache is per database, it should be a global config option (i.e. not per repository).
You can configure it separately for each db right now, but I just think it's wrong it's asking for the repository config instead of the (db wise) global config.